Disable fenv.h support for uClibc-based toolchains.

The boost build system does not recognize the fact that fenv.h is an
optional module in uClibc and tries to use it even if UCLIBC_HAS_FENV
is disabled. This patch disables fenv support completely when compiling
with a uClibc-based toolchain. Bug was reported upstream:
https://svn.boost.org/trac/boost/ticket/11756

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

--- boost_1_60_0.org/boost/config/platform/linux.hpp	2015-12-08 19:55:19.000000000 +0100
+++ boost_1_60_0/boost/config/platform/linux.hpp	2016-02-06 12:35:25.692754553 +0100
@@ -47,6 +47,13 @@
 #endif
 
 //
+// uClibc has no support for fenv.h
+//
+#if defined(__UCLIBC__)
+#  define BOOST_NO_FENV_H
+#endif
+
+//
 // If glibc is past version 2 then we definitely have
 // gettimeofday, earlier versions may or may not have it:
 //
